pkgs/games/uhexen2/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib, fetchgit, SDL, stdenv, libogg, libvorbis, libmad, xdelta }:
+
+stdenv.mkDerivation rec {
+  name = "uhexen2";
+  version = "1.5.9";
+
+  src = fetchgit {
+    url = "https://git.code.sf.net/p/uhexen2/uhexen2";
+    sha256 = "0crdihbnb92awkikn15mzdpkj1x9s34xixf1r7fxxf762m60niks";
+    rev = "4ef664bc41e3998b0d2a55ff1166dadf34c936be";
+  };
+
+  buildInputs = [ SDL libogg libvorbis libmad xdelta ];
+
+  buildPhase = ''
+    make -C engine/hexen2 glh2
+    make -C engine/hexen2 clean
+    make -C engine/hexen2 h2
+    make -C engine/hexen2/server
+
+    make -C engine/hexenworld/client glhw
+    make -C engine/hexenworld/client clean
+    make -C engine/hexenworld/client hw
+    make -C engine/hexenworld/server
+
+    make -C h2patch/
+  '';
```

Comment:
Can this be simplified? Can we use preBuild and makeFlags instead? Also it is missing -j$NIX_BUILD_CORES.

Reply:
Tried to modify it such that the directories containing the makefiles, and their specific arguments, are declared in preBuild. I also changed the buildPhase to be more similar to the one within stdenv. However, I don't think it's particularly more simple. What do you think of the new approach?